src/lib.rs

```diff
@@ -156,50 +156,67 @@ pub fn uring_builder() -> io_uring::Builder {
 ///
-/// Refer to Builder::start for an example.
+/// Refer to [`Builder::start`] for an example.
 pub fn builder() -> Builder {
     Builder {
-        entries: 256,
+        sq_entries: 256,
+        cq_entries: 512,
         urb: io_uring::IoUring::builder(),
     }
 }
 
 impl Builder {
-    /// Set number of submission queue entries in uring.
+    /// Sets the number of submission queue entries in uring. The default is 256.
     ///
-    /// The kernel will ensure it uses a power of two and will round this up if necessary.
-    /// The kernel requires the number of completion queue entries to be larger than
-    /// the submission queue entries so generally will double the sq entries count.
+    /// The kernel requires the number of submission queue entries to be a power of two,
+    /// and that it be less than the number of completion queue entries.
+    /// This function will adjust the `cq_entries` value to be at least 2 times `sq_entries`
+    pub fn sq_entries(&mut self, sqe: u32) -> &mut Self {
+        self.sq_entries = sqe.next_power_of_two();
+        if self.cq_entries / 2 < self.sq_entries {
+            self.cq_entries = self.sq_entries * 2;
+        }
+        self.urb.setup_cqsize(self.cq_entries);
+        self
+    }
+
+    /// Sets the number of completion queue entries in uring. The default is 512.
     ///
-    /// The caller can specify even a larger cq entries count by using the uring_builder
-    /// as shown in the start example below.
-    pub fn entries(&mut self, e: u32) -> &mut Self {
-        self.entries = e;
+    /// The kernel requires the number of completion queue entries to be a power of two,
+    /// and that it be greater than the number of submission queue entries.
+    /// This function will adjust the `cq_entries` value to be at least 2 times `sq_entries`
+    pub fn cq_entries(&mut self, cqe: u32) -> &mut Self {
+        self.cq_entries = cqe.next_power_of_two();
+        if (self.cq_entries / 2) < self.sq_entries {
+            self.sq_entries = self.cq_entries / 2;
+        }
+        self.urb.setup_cqsize(self.cq_entries);
         self
     }
 
-    /// Replace the default io_uring Builder. This allows the caller to craft the io_uring Builder
-    /// using the io_uring crate's Builder API.
+    /// Replaces the default [`io_uring::Builder`], which controls the settings for the
+    /// inner `io_uring` API. This interface offers many more settings and parameters,
+    /// but requires directly depending upon the `io_uring` crate.
     ///
-    /// Refer to the Builder start method for an example.
-    /// Refer to the io_uring::builder documentation for all the supported methods.
+    /// Refer to the [`io_uring::builder`] documentation for all the supported methods.
     pub fn uring_builder(&mut self, b: &io_uring::Builder) -> &mut Self {
         self.urb = b.clone();
         self
     }
 
-    /// Start an `io_uring` enabled Tokio runtime.
+    /// Starts an `io_uring` enabled Tokio runtime.
```

Comment:
above on line 215, the self.urb gets replaced with a b.clone(), even if the user had specified the other cq parameter first. The proposed api looks easier but without more explanation, it can go wrong.